components/cosmetic_filters/resources/data/procedural_filters.ts

```diff
@@ -0,0 +1,607 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+/**
+ *
+ *  This file should be kept up to date with
+ *  https://github.com/brave-experiments/procedural-filters-js
+ *  until the full implementation can be shared with iOS inside
+ *  the brave-core repo.
+ *
+ */
+
+/**
+ *
+ * src/declarations.d.ts
+ *
+ * note: `ProceduralOperator` and `ProceduralSelector`
+ *       are in components/definitions/global.d.ts
+ *
+ */
+
+type CSSSelector = string
+type CSSInstruction = string
+type CSSValue = string
+
+type OperatorType = string
```

Comment:
SonarCloud has problems with these type aliases.

@pes10k and I agreed that they improve the readability of this script, and should be left as-is ([discussion](https://bravesoftware.slack.com/archives/C7VLGSR55/p1722466782261639)).